### PR TITLE
Upgrade flow

### PR DESCRIPTION
**Pull request by ahuth**
Associated Issue: #3683 

### Summary of Changes

* Updated Flow to 0.53.0.
* Moved `defaultProps` for class components to `static` properties, which Flow understands.
* Updated the syntax of the flow typing. Note that I have not tried to improve on any of the type signatures at all.

In particular, instead of defining classes like

```js
class Foo extends Component {
  props: Props
  state: State
}
```

it will now be

```js
class Foo extends Component<Props, State>
```

### Test Plan

* Tests
* Manual QA

### CODE REVIEW

File: src/components/shared/Modal.js [hunk 1]
```diff
@@ -1,23 +1,20 @@
 // @flow
 
 import PropTypes from "prop-types";
-import React, { Component } from "react";
-import type { Children } from "react";
+import * as React from "react";
```
Comment (ahuth) [inline src/components/shared/Modal.js]: This is required to make use of the `React.Node` or `React.Element` types. Alternatively, we could do something like

```js
import type { Node } from "react"
```

**Comment by ahuth** [comment]
Right now I'm getting flow warnings for `node_modules/devtools-reps/src/object-inspector/index.js`. We may need to update that library, as well.

File: src/components/shared/Autocomplete.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  items: Array<Object>,
+  close: (value: any) => void,
+  inputValue: string,
+  placeholder: string,
+  size: string,
+  onChangeHandler: (queryString: string) => void,
+  children: any
+};
+
+export default class Autocomplete extends Component<Props, State> {
```
Comment (ahuth) [inline src/components/shared/Autocomplete.js]: Uh oh, something is up with this diff. Something may have gone wrong while merging in master from upstream.

**Comment by jasonLaster** [comment]
@ahuth just read through it and it looks great! Thank you so much for your hard work

**Comment by jasonLaster** [comment]
@ahuth feel free to disable flow checks for it in `flow_config` I want this to land :)

it's too valuable

**Comment by ahuth** [comment]
@jasonLaster, this should be good to go, although I haven't manually QA'd it, yet.

Upgraded to 0.53.0. Going farther than that will take some more work. I'll do that sometime, and may also come back and cleanup some of the types.

**Comment by codehag** [comment]
this looks good

**Comment by ahuth** [comment]
Thanks

File: src/components/shared/Modal.js (hunk 1 above)
Comment (jasonLaster) [reply src/components/shared/Modal.js]: I like this

**Comment by jasonLaster** [comment]
Thanks for the help! You should stop by our [slack](https://devtools-html-slack.herokuapp.com) :wave:

**Pull request merged**